nilearn/decoding/space_net.py

```diff
@@ -0,0 +1,1242 @@
+        self.n_jobs = n_jobs
+        self.cv = cv
+        self.screening_percentile = screening_percentile
+        self.debias = debias
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+
+        # sanity check on params
+        self.check_params()
+
+    def check_params(self):
+        """Makes sure parameters are sane"""
+        if not self.l1_ratios is None:
+            l1_ratios = self.l1_ratios
+            if isinstance(l1_ratios, numbers.Number):
+                l1_ratios = [l1_ratios]
+            for l1_ratio in l1_ratios:
+                if not 0 <= l1_ratio <= 1.:
+                    raise ValueError(
+                        "l1_ratio must be in the interval [0, 1]; got %g" % (
+                            l1_ratio))
+                elif l1_ratio == 0. or l1_ratio == 1.:
+                    warnings.warn(
+                        ("Specified l1_ratio = %g. It's adived to only "
+                         "specify values of l1_ratio strictly between 0 "
+                         "and 1." % l1_ratio))
+        if not (0. <= self.screening_percentile <= 100.):
+            raise ValueError(
+                ("screening_percentile should be in the interval"
+                 " [0, 100], got %g" % self.screening_percentile))
+        if self.penalty not in self.SUPPORTED_PENALTIES:
+            raise ValueError(
+                "'penalty' parameter must be one of %s%s or %s; got %s" % (
+                    ",".join(self.SUPPORTED_PENALTIES[:-1]), "," if len(
+                        self.SUPPORTED_PENALTIES) > 2 else "",
+                    self.SUPPORTED_PENALTIES[-1], self.penalty))
+        if not (self.loss is None or self.loss in self.SUPPORTED_LOSSES):
+            raise ValueError(
+                "'loss' parameter must be one of %s%s or %s; got %s" % (
+                    ",".join(self.SUPPORTED_LOSSES[:-1]), "," if len(
+                        self.SUPPORTED_LOSSES) > 2 else "",
+                    self.SUPPORTED_LOSSES[-1], self.loss))
+        if not self.loss is None and not self.is_classif and (
+            self.loss == "logistic"):
+            raise ValueError(
+                ("'logistic' loss is only available for classification "
+                 "problems."))
+
+    def _set_coef_and_intercept(self, w):
+        """Sets the loadings vector (coef) and the intercept of the fitted
+        model."""
+        self.w_ = np.array(w)
+        if self.w_.ndim == 1:
+            self.w_ = self.w_[np.newaxis, :]
+        self.coef_ = self.w_[:, :-1]
+        if self.is_classif:
+            self.intercept_ = self.w_[:, -1]
+        else:
+            self._set_intercept(self.Xmean_, self.ymean_, self.Xstd_)
+
+    def fit(self, X, y):
+        """Fit the learner
+
+        Parameters
+        ----------
+        X : list of filenames or NiImages of length n_samples, or 2D array of
```

Comment:
Does it really work with np arrays? Doesn't TV require an affine (and therefore) a Niimage? Doesn't the
 _univariate_feature_screening also require an affine to estimate the brain volume? Anyway the fit method appears to require a NiftiMasker that fit_transforms the X argument. Which would fail if X is an array.